### PR TITLE
Interpolation method as a toggle

### PR DESCRIPTION
**Pull request by samcunliffe**
Following on from #196 & #198 weird wrong diff bug.

### CODE REVIEW

**Review by willGraham01 — APPROVED** [review]
Looks amazing, but oddly familiar...

File: tdms/include/field.h [hunk 1]
```diff
@@ -25,18 +25,26 @@
...
      * Maximum value out of I_tot, J_tot and K_tot
      * @return value
      */
-    int max_IJK_tot() const {return max(I_tot, J_tot, K_tot); };
+  int max_IJK_tot() const { return max(I_tot, J_tot, K_tot); };
+
+  /**
+   * @brief Set the preferred interpolation methods
+   */
+  void set_pim(PreferredInterpolationMethods _pim) { pim = _pim; };
```
Comment (samcunliffe) [inline tdms/include/field.h:47]:
```suggestion
  void set_preferred_interpolation_methods(PreferredInterpolationMethods _pim) { pim = _pim; };
```

File: tdms/include/globals.h [hunk 2]
```diff
@@ -53,6 +53,7 @@ enum AxialDirection
 enum ModeOfRun    { Pass1 , Pass2 };
 enum RCSType      { parallel , perpendicular };
 enum SolverMethod { PseudoSpectral, FiniteDifference };
+enum PreferredInterpolationMethods { BandLimited, Cubic };
```
Comment (samcunliffe) [inline tdms/include/globals.h:56]:
```suggestion
enum PreferredInterpolationMethod { BandLimited, Cubic };

```
Why not singular? Do I miss something?

File: tdms/src/argument_parser.cpp [hunk 3]
```diff
@@ -51,7 +51,8 @@ void ArgumentParser::print_help_message(){
                  "openandorder [options] infile gridfile outfile\n"
                  "Options:\n"
                  "-h:\tDisplay this help message\n"
-                 "--finite-difference:\tUse the finite-difference solver, instead of the pseudo-spectral method.\n"
+                 "-fd, --finite-difference:\tUse the finite-difference solver, instead of the pseudo-spectral method.\n"
+                 "-nbli, --no-band-limited:\tDo not use band-limited interpolation to determine field values at Yee cell centres."
```
Comment (samcunliffe) [inline tdms/src/argument_parser.cpp:55]: I might have gone for `-c, --cubic-interpolation` and have it off by default.

**Comment by willGraham01** [comment]
Merging into #151 as failing tests are due to interpolation method changes, rather than other errors.

**Pull request merged**